test/Microsoft.ApplicationInsights.AspNet.Tests/ApplicationInsightsExceptionMiddlewareTest.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Microsoft.ApplicationInsights.AspNet
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.AspNet.Tests;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.AspNet.Builder;
+    using Xunit;
+
+    public class ApplicationInsightsExceptionMiddlewareTest
+    {
+        private ITelemetry sentTelemetry;
+
+        [Fact]
+        public async Task InvokeTracksExceptionThrownByNextMiddlewareAsHandledByPlatform()
+        {
+            RequestDelegate nextMiddleware = httpContext => { throw new Exception(); };
+            var middleware = new ApplicationInsightsExceptionMiddleware(nextMiddleware, MockTelemetryClient());
+
+            await Assert.ThrowsAnyAsync<Exception>(() => middleware.Invoke(null));
+
+            Assert.Equal(ExceptionHandledAt.Platform, ((ExceptionTelemetry)sentTelemetry).HandledAt);
+        }
+
+        private TelemetryClient MockTelemetryClient()
+        {
```

Comment:
This is a lot of code to mock a TelemetryChannel. It would be nice to have an interface with a single Track(ITelemetry) method.